### PR TITLE
impl Error for CompoundTagError

### PR DESCRIPTION
**Pull request by SpecificProtagonist**
`CompoundTagError` and `TagDecodeError` were missing their `std::error::Error` impl.

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -2,8 +2,8 @@ use crate::{CompoundTag, Tag};
 use byteorder::{BigEndian, ReadBytesExt};
 use flate2::read::{GzDecoder, ZlibDecoder};
 use linked_hash_map::LinkedHashMap;
-use std::io;
-use std::io::Read;
+use std::{error::Error, io::Read};
```
Comment (vaIgarashi) [inline src/decode.rs:5]: Can you pls format with default formatting rules?

File: src/decode.rs [hunk 2]
```diff
@@ -28,6 +28,30 @@ impl From<io::Error> for TagDecodeError {
...
+        match self {
+            TagDecodeError::IOError { io_error } => Some(io_error),
+            _ => None,
+        }
+    }
+}
+
+impl Display for TagDecodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use TagDecodeError::*;
```
Comment (vaIgarashi) [inline src/decode.rs]: Why you put this import here?

File: src/lib.rs [hunk 3]
```diff
@@ -212,6 +215,23 @@ pub enum CompoundTagError<'a> {
     },
 }
 
+impl<'a> std::error::Error for CompoundTagError<'a> {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        None
```
Comment (vaIgarashi) [inline src/lib.rs:220]: Ain't here should be provided all enum variants with their variables?

**Review by vaIgarashi — CHANGES_REQUESTED** [review]
Some minor changes required

File: src/lib.rs (hunk 3 above)
Comment (SpecificProtagonist) [reply src/lib.rs:220]: `source` is for nested errors, of which there are none.

File: src/decode.rs (hunk 2 above)
Comment (SpecificProtagonist) [reply src/decode.rs]: Just for the match; I've removed it and repushed.

File: src/decode.rs (hunk 1 above)
Comment (SpecificProtagonist) [reply src/decode.rs:5]: Hm, I think I'm using the default settings. How does the result of rustfmt differ for you?

**Review by vaIgarashi — APPROVED** [review]

**Pull request merged**